components/region_cache_memory_engine/src/engine.rs

```diff
@@ -132,39 +221,84 @@ impl Iterable for RegionCacheMemoryEngine {
 
 impl Iterator for RegionCacheIterator {
     fn key(&self) -> &[u8] {
-        unimplemented!()
+        assert!(self.valid);
+        self.iter.key().as_slice()
     }
 
     fn value(&self) -> &[u8] {
-        unimplemented!()
+        assert!(self.valid);
+        self.iter.value().as_slice()
     }
 
     fn next(&mut self) -> Result<bool> {
-        unimplemented!()
+        assert!(self.valid);
+        self.iter.next();
+        self.valid = self.iter.valid() && self.iter.key().as_slice() < self.upper_bound.as_slice();
+
+        if self.valid && self.prefix_same_as_start {
+            // todo(SpadeA): support prefix seek
+            unimplemented!()
+        }
+        Ok(self.valid)
     }
 
     fn prev(&mut self) -> Result<bool> {
-        unimplemented!()
+        assert!(self.valid);
+        self.iter.prev();
+        self.valid = self.iter.valid() && self.iter.key().as_slice() >= self.lower_bound.as_slice();
+        if self.valid && self.prefix_same_as_start {
+            // todo(SpadeA): support prefix seek
+            unimplemented!()
+        }
+        Ok(self.valid)
     }
 
     fn seek(&mut self, key: &[u8]) -> Result<bool> {
-        unimplemented!()
+        let seek_key = if key < self.lower_bound.as_slice() {
```

Comment:
Do it needs to check ` assert!(self.valid);` before this line?

Reply:
seek deos not depend on the previous operations, so it does not need it.